app/controllers/admin/enterprises_controller.rb

```diff
@@ -316,15 +316,22 @@ def enterprise_params
       return params[:enterprise] if params[:enterprise].empty?
 
       params.require(:enterprise).permit(
-        :name, :is_primary_producer, :visible, :permalink,
+        :id, :name, :is_primary_producer, :visible, :permalink,
         :contact_name, :email_address, :phone, :sells, :owner_id,
         :website, :facebook, :instagram, :linkedin, :twitter,
         :abn, :acn, :charges_sales_tax, :display_invoice_logo,
         :invoice_text, :description, :long_description, :promo_image,
         :preferred_product_selection_from_inventory_only, :preferred_shopfront_message,
         :preferred_shopfront_closed_message, :preferred_shopfront_taxon_order,
         :preferred_shopfront_order_cycle_order, :require_login,
-        :allow_guest_orders, :allow_order_changes, :enable_subscriptions)
+        :allow_guest_orders, :allow_order_changes, :enable_subscriptions,
+        address_attributes: permitted_address_attributes
+      )
+    end
+
+    # Used in ResourceController#create
+    def permitted_resource_params
+      enterprise_params
```

Comment:
Looking at this I'm wondering if it would be DRYer if we just define `permitted_resource_params` in each of our controllers, and call that in the methods where we use params, instead of defining two methods everywhere...

Reply:
yeah, I see, I prefer to keep it like this so that resource controller stuff is not the default. I think we should try to remove resource controller all together with its "callbacks and stuff.."
I like the default as enterprise_params.

Reply:
yeah, I see, I prefer to keep it like this so that resource controller stuff is not the default. I think we should try to remove resource controller all together with its "callbacks and stuff.."
I like the default as enterprise_params.